tests/cypress/support/commands.js

```diff
@@ -278,3 +280,44 @@ Cypress.Commands.add("getMany", (names) => {
     }
     return cy.wrap(values);
 });
+
+Cypress.Commands.add("createWithAPI", (url, values) => {
+    return cy.initApi().doApiRequest("POST", url, values).then(response => {
+        return response.body.id;
+    });
+});
+
+Cypress.Commands.add("updateWithAPI", (url, values) => {
+    cy.initApi().doApiRequest("PUT", url, values);
+});
+
+Cypress.Commands.add("initApi", () => {
+    if (api_token !== null) {
+        return api_token;
+    }
+
+    cy.request({
+        auth: {
+            'user': 'e2e_tests',
+            'pass': 'glpi',
+        },
+        method: 'POST',
+        url: '/apirest.php/initSession',
+    }).then((response) => {
+        return response.body.session_token;
+    });
+});
+
+Cypress.Commands.add("doApiRequest", {prevSubject: true}, (token, method, endpoint, values) => {
+    cy.request({
+        method: method,
+        url: '/apirest.php/' + endpoint,
+        body: {input: values},
+        headers: {
+            'Session-Token': token,
+            'GLPI-Profile': 4
```

Comment:
Wrong API. Legacy API forces you to use a session and doesn't let you choose entity or profile on a per-request basis.

Reply:
I am not against using the new API but as far as I know it is suppose to be a "high level" API.

I don't know much about it yet but I would expect a "high level" API to be limited and allow less liberty than the "low level" API (which would then be more adapted for this use case).

Is this not the case ?

Reply:
I'm not saying *don't* use the legacy API. I am just saying that using a `GLPI-Profile` header with it isn't doing anything because that API doesn't use it. I don't have a strong opinion that favors one API over the other.

They both have drawbacks, but most of the drawbacks of the legacy API disappear when it isn't a human (at least one that isn't familiar with the internals of GLPI) using it.

Reply:
Also, we could try to combine tests into the same spec when possible. For example, the timeline actions aren't very complicated features. They can be combined into a single spec (but different tests) with a single ticket being created in the `before` hook and then sharing that ticket ID/url with the individual tests.

Reply:
I see, I didn't realize you where talking about the `GLPI-Profile': 4` line, which is indeed a mistake.

Fixed it, thanks.